_posts/2024-09-27-learning-a-programming-language-with-generative-ai.md

```diff
@@ -0,0 +1,58 @@
+
+Now and then it generated not working and non-coherent code. Because of that I could not rely solely on ChatGPT but also had to google certain subjects as well as reading code documentation.
+
+In the end, however, the user story was implemented as it was supposed to be, although the implementation might not have been the best possible.
+
+# The learning experience
+
+Now and then, talking with ChatGPT might feel like having a conversation with a real person. Something that has come to my mind more than once is the Turing test. For a recap, it states that a machine can be seen as intelligent as a human, if it can engage in a conversation without being detected as a machine.  I would say that the test might now be considered as succeeded. ChatGPT can engage in a conversation by answering questions. The answers are usually grammatically correct and remind of human generated text.  And like when asking questions from a real person, the answers are not always true. The risk with a succeeding Turing test is that one might have the illusion of conversing with a human but forget that humans make mistakes. On the other hand, as it is not asking any own questions, a discussion with ChatGPT may not be a conversation in the traditional form. 
+
+![The turing tes](/img/2024-09-27-learning-a-programming-language-with-generative-ai/turing-test.png)
+
+# How to learn with GenAI
+
+Myself, I used a learning by doing approach. There are however, many ways one can learn with GenAI. Another way to learn a new programming language could be to step by step letting ChatGPT talk about the language in question, including its syntax, variables, typing, data structures and so on. I, however, would not find this approach useful as there are already online-documentation and books for most programming languages that can be used in the traditional way, and that are also reliable. Regardless of the way, there are some practices that are good to take into consideration:
+
+- ChatGPT is probably the best option for learning. AI-tools for coding do not explain and tell theory and concepts as well as ChatGPT
+- If using ChatGPT, it is preferable to use the paid version and create a customized agent. That will be trained to focus as much as possible on the subject in question.
+- Fact check, fact check, fact check! GenAI is not all knowing and will make mistakes!
+- Do not over rely on GenAI. When the learning seems to be going well, there is always the risk that one will trust it too much. That might lead to forgetting of fact checking as well as passivity and decreased learning, as GenAI is doing most of the job.
```

Comment:
"That might lead to forgetting of fact checking as well as passivity and decreased learning" -> I find this hard to understand. Could is be something like "Too much trust might cause you to forget to fact check as well as causing passivity and decreased learning." Idk, that's a difficult sentence

Reply:
corrected